yarn-project/foundation/src/abi/encoder.ts

```diff
@@ -29,9 +29,13 @@ class ArgumentEncoder {
         } else if (typeof arg === 'object') {
           if (typeof arg.toField === 'function') {
             this.flattened.push(arg.toField());
-          } else {
+          } else if (arg instanceof Fr) {
             this.flattened.push(arg);
+          } else {
+            throw new Error('Object cannot be serialized to a field and is not a field.');
```

Comment:
I think I would just print 'Argument cannot be serialised to a field'.

Reply:
Addressed in [25d1ad1](https://github.com/AztecProtocol/aztec-packages/pull/1246/commits/25d1ad1833260de0654cc9250ef1a4e9e2159f49)